File: Assembly-FSharp/Assembly-FSharp/Commons/Casters/SphereCast.fs
namespace Styx.Commons.Casters

open UnityEngine

[<CreateAssetMenu(fileName = "Pass Cast", menuName = "Casts/Pass Cast")>]
type SphereCast () =
    inherit ScriptableObject()

    [<SerializeField>]
    let mutable radius: float32 = 0.f
    [<SerializeField>]
    let mutable maxDistance: float32 = 0.f
    [<SerializeField>]
    let mutable mask: LayerMask = Unchecked.defaultof<LayerMask>
    [<SerializeField>]
    let mutable originOffset: Vector3 = Vector3.zero

    member this.Radius
        with get() = radius
    member this.MaxDistance
        with get() = maxDistance
    member this.Mask
        with get() = mask
    member this.OriginOffset
        with get() = originOffset

File: Assets/_Scripts/Commons/Casters/SphereCast.cs
using UnityEngine;

namespace RageCure.Commons.Casters
{
    [CreateAssetMenu(fileName = "Pass Cast", menuName = "Casts/Pass Cast")]
    public class SphereCast : ScriptableObject
    {
        [SerializeField]
        private float _radius = 0f;
        [SerializeField]
        private float _maxDistance = 0f;
        [SerializeField]
        private LayerMask _mask;
        [SerializeField]
        private Vector3 _originOffset = Vector3.zero;

        public float Radius { get { return _radius; } }
        public float MaxDistance { get { return _maxDistance; } }
        public LayerMask Mask { get { return _mask; } }
        public Vector3 OriginOffset { get { return _originOffset; } }
    }
}
